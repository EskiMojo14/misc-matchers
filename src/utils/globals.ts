// to avoid having separate jest and vitest suites, we create common definitions
// then mock during setup with the appropriate globals
import type {
  describe as jestDescribe,
  it as jestIt,
  expect as jestExpect,
  jest,
} from "@jest/globals";
import type {
  describe as vitestDescribe,
  it as vitestIt,
  expect as vitestExpect,
  vi,
  Mock as VitestMock,
} from "vitest";
import type { Satisfies } from "./types.typetest";

interface Each {
  <T extends Array<any> | [any]>(
    cases: ReadonlyArray<T>,
  ): (name: string, fn: (...args: T) => void) => void;
  <T>(cases: ReadonlyArray<T>): (name: string, fn: (value: T) => void) => void;
}
export declare const describe: (typeof vitestDescribe | typeof jestDescribe) & {
  each: Each;
};

// eslint-disable-next-line @typescript-eslint/no-unused-vars
type TestEach = Satisfies<
  typeof jestDescribe.each | typeof vitestDescribe.each,
  Each
>;

export declare const expect: typeof vitestExpect | typeof jestExpect;
export declare const it: typeof vitestIt | typeof jestIt;

export type Mock<R = any, A extends Array<any> = Array<any>, T = any> =
  | jest.Mock<(this: T, ...args: A) => R>
  | VitestMock<A, R>;

export declare const fn: {
  (): Mock;
  <R, A extends Array<any>, T>(fn?: (this: T, ...args: A) => R): Mock<R, A, T>;
};

// eslint-disable-next-line @typescript-eslint/no-unused-vars
type TestFn = Satisfies<typeof jest.fn | typeof vi.fn, typeof fn>;

throw new Error("This file should be mocked");
