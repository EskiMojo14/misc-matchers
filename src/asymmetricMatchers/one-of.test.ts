import { describe, it, expect } from "@jest/globals";

describe("oneOf", () => {
  it("should pass if value matches one of provided values", () => {
    expect(1).toEqual(expect.oneOf([1, 2, 3]));
    expect(2).toEqual(expect.oneOf([1, 2, 3]));
    expect(3).toEqual(expect.oneOf([1, 2, 3]));
    expect(4).toEqual(expect.not.oneOf([1, 2, 3]));
  });
  it("should fail if value does not match any of the provided values", () => {
    expect(() => {
      expect(4).toEqual(expect.oneOf([1, 2, 3]));
    }).toThrowErrorMatchingSnapshot();
    expect(() => {
      expect(1).toEqual(expect.not.oneOf([1, 2, 3]));
    }).toThrowErrorMatchingSnapshot();
  });
});
