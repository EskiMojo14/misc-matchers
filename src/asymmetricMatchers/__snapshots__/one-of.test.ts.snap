// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`oneOf should fail if value does not match any of the provided values 1`] = `
<d>expect(</b><r>received</g><d>).</b>toEqual<d>(</b><g>expected</g><d>) // deep equality</b>

Expected: <g>oneOf<1,2,3></g>
Received: <r>4</g>
`;

exports[`oneOf should fail if value does not match any of the provided values 2`] = `
<d>expect(</b><r>received</g><d>).</b>toEqual<d>(</b><g>expected</g><d>) // deep equality</b>

Expected: <g>not.oneOf<1,2,3></g>
Received: <r>1</g>
`;
